_posts/2015-04-30-gsoc-15-is-now-live.markdown

```diff
@@ -0,0 +1,123 @@
+improve our asset [re]generation speed. The initial scope of this project covers CoffeeScript, Sass and
+our Uglifier, but benchmarks will be guiding this effort to work where we can get the biggest
+benefits in these four months.
+
+## Evented File System Monitoring
+
+**Student:** [Puneet Agarwal](https://github.com/puneet24)    
+**Mentors:** [Xavier Noria](https://github.com/puneet24) and [Matthew Draper](https://github.com/matthewd)
+
+[FileUpdateChecker](http://api.rubyonrails.org/classes/ActiveSupport/FileUpdateChecker.html), the
+system we use for detecting changes in files (mostly for reloading purposes) has served us well over the years, but we're done with
+polling. Puneet will be replacing our current design with a event-based approach that relies
+on existing third-party monitors (e.g. `inotify` or `FSEvent`).
+
+## Asset Pipeline Support for Source Maps
+
+**Student:** [Andrei Istratii](https://github.com/Andreis13)    
+**Mentors:** [Arthur Nogueira Neves](https://github.com/arthurnn) and [Rafael Mendonça França](https://github.com/rafaelfranca)
+
+The goal of Andrei's project is to give you good inspecting and debugging capabilities in environments
+where your code goes through various transformations (e.g. your CoffeeScript file being compiled to
+Javascript and then minified in your staging environment). With source maps you can use the existing
+tools your browser provides to do things like reading the CoffeeScript source or setting breakpoints
+on it.
+
+## Refactoring ActiveSupport and ActionView
+
+**Student:** [Islam Wazery](https://github.com/wazery)    
+**Mentors:** [Kir Shatrov](https://github.com/kirs) and [Carlos Antonio da Silva](https://github.com/carlosantoniodasilva)
+
+Islam is adding some of the things we should already have in Rails, liked named arguments
+ for ActionView helpers (goodbye counting commas!), or removing the ActiveSupport dependency on
+ ActionPack. He will also be investigating how to improve some of our core abstractions like
+ [ActionController::Parameters](http://edgeapi.rubyonrails.org/classes/ActionController/Parameters.html)
+ and
+ [ActionView::OutputBuffer](http://tenderlovemaking.com/2014/06/04/yagni-methods-slow-us-down.html)
+ to enable better security and performance.
+
+## Web Console Browser Extensions
+
+**Student:** [Hiroyuki Sano](https://github.com/sh19910711)    
+**Mentor**: [Genadi Samokovarov](https://github.com/gsamokovarov)
+
+Following up on the work of previous GSoC projects, Hiroyuki will be creating browser extensions for
+the Rails web console. Like the the Source Maps project, this one will give you a better live
+debugging experience using standard tools that everyone has available already.
+
+## Test Failure Prediction
+
+**Student:** [Genki Sugimoto](https://github.com/Genki-S)    
+**Mentors:** [Robin Dupret](https://github.com/robin850), **TODO** Confirm: "and maybe Josh K from travis CI"
```

Comment:
confirm, sorry, these last several weeks have been a bit crazy due to me moving from NZ to Berlin.